app/actions/views/custom_status.ts

```diff
@@ -1,23 +1,55 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 import {UserCustomStatus} from '@mm-redux/types/users';
-import {ActionFunc} from '@mm-redux/types/actions';
+import {ActionFunc, DispatchFunc, batchActions, GetStateFunc} from '@mm-redux/types/actions';
 import {bindClientFunc} from '@mm-redux/actions/helpers';
 import {Client4} from '@mm-redux/client';
+import {UserTypes} from '@mm-redux/action_types';
+import {logError} from '@mm-redux/actions/errors';
+import {getCurrentUser} from '@mm-redux/selectors/entities/common';
 
 export function setCustomStatus(customStatus: UserCustomStatus): ActionFunc {
-    return bindClientFunc({
-        clientFunc: Client4.updateCustomStatus,
-        params: [
-            customStatus,
-        ],
-    });
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        const user = getCurrentUser(getState());
+        const oldCustomStatus = user?.props?.customStatus;
+        user.props.customStatus = JSON.stringify(customStatus);
```

Comment:
nit - maybe we should check if new custom status is not the same as previous custom status

Reply:
This action wouldn't be dispatched if old and new custom status were same.